src/test/kotlin/calculator/CalculatorTest.kt

```diff
@@ -0,0 +1,84 @@
+package calculator
+
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.Assertions.assertThatExceptionOfType
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.ValueSource
+
+class CalculatorTest {
+    @Test
+    fun `더하기만 테스트`() {
+        val stringCalculator1 = StringCalculator("1 + 3")
+        val result1 = stringCalculator1.result
+        assertThat(result1).isEqualTo((1 + 3).toDouble())
+
+        val stringCalculator2 = StringCalculator("15 + 31")
+        val result2 = stringCalculator2.result
+        assertThat(result2).isEqualTo((15 + 31).toDouble())
+    }
+
+    @Test
+    fun `빼기만 테스트`() {
+        val stringCalculator1 = StringCalculator("1 - 3")
+        val result1 = stringCalculator1.result
+        assertThat(result1).isEqualTo((1 - 3).toDouble())
+
+        val stringCalculator2 = StringCalculator("15 - 31")
+        val result2 = stringCalculator2.result
+        assertThat(result2).isEqualTo((15 - 31).toDouble())
+    }
+
+    @Test
+    fun `나누기 테스트`() {
+        val stringCalculator1 = StringCalculator("1 / 3")
+        val result1 = stringCalculator1.result
+        assertThat(result1).isEqualTo((1 / 3.0))
+
+        val stringCalculator2 = StringCalculator("15 / 31")
+        val result2 = stringCalculator2.result
+        assertThat(result2).isEqualTo((15 / 31.0))
+    }
+
+    @Test
+    fun `곱하기 테스트`() {
+        val stringCalculator1 = StringCalculator("1 * 3")
+        val result1 = stringCalculator1.result
+        assertThat(result1).isEqualTo((1 * 3).toDouble())
+
+        val stringCalculator2 = StringCalculator("15 * 31")
+        val result2 = stringCalculator2.result
+        assertThat(result2).isEqualTo((15 * 31).toDouble())
+    }
+
+    @Test
+    fun `조합 연산 테스트`() {
+        val stringCalculator = StringCalculator("1 * 3 + 10 / 3 - 5")
+        val result1 = stringCalculator.result
+        var expectedResult: Double = 1 * 3.0
+        expectedResult += 10
+        expectedResult /= 3
+        expectedResult -= 5
+        assertThat(result1).isEqualTo(expectedResult)
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = ["", " ", "\n"])
+    fun `빈칸 테스트`(inputString: String) {
+        assertThatExceptionOfType(IllegalArgumentException::class.java)
+            .isThrownBy {
+                val stringCalculator = StringCalculator(inputString)
```

Comment:
```suggestion
        assertThrows<IllegalArgumentException> { StringCalculator(inputString) }
```

assertThrows를 사용하면, 예외 처리에 대해서도 간결하게 테스트 코드를 짤 수가 있습니다.

assertThrows와 assertThatExceptionOfType의 차이점은 무엇일까요? 🤔

Reply:
추가적인 체크로직을 담을 수 없네요. 흔한 예외타입 체크나, 메시지체크 등을 하지않아도 되는 경우에 쓰면 될거같습니다.

Reply:
해당 부분에 답을 못 드렸네요!
 assertTatExceptionOfType()과 assertThrows<>의 가장 큰 차이점은 .class.java 파일을 쓸 것인지 .kt 파일을 쓸 것인지에 대한 차이점이 있습니다. 이러한 차이점이 있기 때문에 assertThatExceptionOfType은 java Exception class 만 받을 수 있습니다. 코틀린 Exception class를 받기 위해서도 결국 java로 변경해서 사용을 해야될거에요. 
 하지만, assertThrows<>는 이미 .kt 파일로 만들어졌기 때문에, kotlin의 익셉션 처리를 할 수 있고, 순수한 kotiln API로 에러 테스트를 할 수 있게 만듭니다.
 
 저의 개인적인 생각이지만, kotlin 코틀린스럽게 짜는 코드는 kotlin 프로젝트에서 자바와 혼용해서 짜는 것이 아닌, kotlin에 있는 API만으로 코드를 짜는 것부터 시작한다고 생각합니다 : )
 
 앞으로 있을 미션에 대해 java API가 아닌, kotlin API로 짜는 연습을 해보시길 추천드립니다! 😁